core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -386,17 +386,21 @@ class StandaloneDynamicAllocationSuite
     // the driver refuses to kill executors it does not know about
     syncExecutors(sc)
     val executors = getExecutorIds(sc)
+    val executorIdsBefore = executors.head
     assert(executors.size === 2)
     // kill executor 1, and replace it
     assert(sc.killAndReplaceExecutor(executors.head))
     eventually(timeout(10.seconds), interval(10.millis)) {
       val apps = getApplications()
       assert(apps.head.executors.size === 2)
+      val executorIdsAfter = getExecutorIds(sc).head
+      // make sure the old executors head has been killedAndReplaced.
+      assert(executorIdsBefore != executorIdsAfter)
     }
 
     var apps = getApplications()
-    // kill executor 1
-    assert(sc.killExecutor(executors.head))
+    // kill executor 1, and actually nothing to kill
+    assert(!sc.killExecutor(executors.head))
     apps = getApplications()
     assert(apps.head.executors.size === 2)
```

Comment:
since we didn't change the `executors` (essentially it is the `executorIdsBefore`)  after the previous assignment. it is still {27}.  `executors(1) == 28`.